silverback/application.py

```diff
@@ -75,7 +74,55 @@ def __init__(self, settings: Optional[Settings] = None):
 
     def on_startup(self) -> Callable:
         """
-        Code to execute on worker startup / restart after an error.
+        Code to execute on one worker upon startup / restart after an error.
+
+        Usage example::
+
+            @app.on_startup()
+            def do_something_on_startup(startup_state):
+                ...  # Reprocess missed events or blocks
+        """
+        return self.broker.task(task_name="silverback_startup")
+
+    def on_shutdown(self) -> Callable:
+        """
+        Code to execute on one worker at shutdown.
+
+        Usage example::
+
+            @app.on_shutdown()
+            def do_something_on_shutdown():
+                ...  # Record final state of app
+        """
+        return self.broker.task(task_name="silverback_shutdown")
+
+    def on_client_startup(self) -> Callable:
```

Comment:
Was thinking that the client-side logic is fully fulfilled by the `SilverbackApp` class (and thus should not be exposed directly to the user) so keeping this as worker-only startup logic may be simpler, and shouldn't impact us too much down the road because of this decision to own the client-side of the app via the SDK

Reply:
Are you suggesting we remove them?  I'm okay with that, I just left these in to be feature complete.  I didn't document them because they're confusing to users.

Reply:
hmm, just trying to keep it simpler